manifests/plugin/pulp.pp

```diff
@@ -19,6 +19,8 @@
 #
 # $client_authentication:: An array of client authentication types supported by the Pulp installation.
 #
+# $rhsm_url::              The RHSM base URL represents the URL to the Katello API for RHSM traffic, or a reverse proxy to it. Katello will direct hosts registering through this SmartProxy server to send subscription-manager and other RHSM client traffic to this URL.
```

Comment:
To be explicit in what I think is important, how about this?
```suggestion
# $rhsm_url::              RHSM is used by subscription-manager and other RHSM clients. This URL is used by Katello to configure hosts, for example during host registration. It should either be the the URL to the Katello API for RHSM traffic, or a reverse proxy to it. 
```

Reply:
Thinking about this some more:

`This URL is used by Katello to configure hosts, for example during host registration` this is true, but potentially confusing since a lot of users are used to the registration process like :

```shell
yum install http://smartproxy.example.com/pub/katello-ca-consumer-latest.noarch.rpm
subscription-manager register
```

In practice you would use one or more activation keys as well but that does a lot more than just _registration_ with RHSM. Even the minimal version above requires you to specify the Organization, LCE, etc. (to @ehelms prior point about Katello's special handling of /rhsm)

Anyway, if you think of registration as happening during the `subscription-manager register` command, then it feels wrong to say that configuring RHSM clients to use the RHSM URL happens during registration.

If instead you think of registration as starting with installing the bootstrap RPM, or using WebUI/hammer/API/FAM to generate the script to configure & register your host, then this wording makes more sense.

As an aside, I am kind of skeptical about global registration as the preferred way to register hosts because it requires the user to interact with Katello, whereas the other method requires only a simple `yum install ...` and `subscription-manager register` with sufficiently configured activation keys. So the Katello admin's end users (who can often be external to their team, department, or even organization) don't require any Katello knowledge nor any login in order to do it -- just the URL for the bootstrap RPM and the (list of) activation keys they've been provided to use for, e.g. an EL8 database server in the production environment.

Reply:
Of course the end users who register to Katello without needing to know anything about Katello, are not the same ones that are going to be reading doc strings for these puppet module parameters.

So maybe the clearest thing is to be specific that the `configure hosts` part happens when either the bootstrap RPM is installed or the global registration templates are rendered and then executed on the target system.

Reply:
The other potential area for concern I have is that `It should either be the the URL to the Katello API for RHSM traffic, or a reverse proxy to it` is that it's not immediately clear without stopping to think about it, that this URL needs to be accessible to the hosts who will register through this particular SmartProxy instance.

If I have Katello, and an additional external SmartProxy that serves hosts on an isolated network, then configuring that SmartProxy with Katello's RHSM URL is not going to work.

So I think we should make it explicit, without relying on context of knowing "what does this module do?", that this URL is the one which will be used by hosts configured to register via this particular SmartProxy instance.

Reply:
I am worried we are splitting hairs on this documentation and over thinking it, trying to drive too deep into the various functionalities that might exist. Plain and simple, this is the API Katello uses for RHSM traffic, and sometimes its deployed as a reverse proxy to the server serving the actual API. There are too many ways that hosts might use this, or states hosts might be in when using this to capture them all. There are myriad ways a host could be configured to talk to this endpoint. 

 This parameter won't be exposed to end users of the foreman-installer due to how we wrap the module, and generally will be set programmatically as setting this to something other than what Katello's actual API is will result in a most likely non-functioning setup. The docs are also non-binding, and easy to update if we feel we need further clarification. That does not change the implementation or what problem this is working to help solve. 

I'd vote let's go with what's here.

Reply:
> The other potential area for concern I have is that `It should either be the the URL to the Katello API for RHSM traffic, or a reverse proxy to it` is that it's not immediately clear without stopping to think about it, that this URL needs to be accessible to the hosts who will register through this particular SmartProxy instance.

I think that's covered by "RHSM is used by subscription-manager and other RHSM clients". To me it's implied that if those clients need to use it, they need to connect with it.

Reply:
> This parameter won't be exposed to end users of the foreman-installer due to how we wrap the module, and generally will be set programmatically as setting this to something other than what Katello's actual API is will result in a most likely non-functioning setup. The docs are also non-binding, and easy to update if we feel we need further clarification. That does not change the implementation or what problem this is working to help solve.

If we follow that logic, we can also leave out the documentation.

Reply:
I think I am OK with this text now. Could you make it `Smart Proxy` and wrap the text to new lines where it gets long?